### PR TITLE
Task0 unit tests

### PR DESCRIPTION
**Pull request by GDP254**
Hi, Please review my task0 unit tests.

### CODE REVIEW

File: fellow_tests.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		fellow.save_state()
+		fellow_1 = Fellow("LIOZ", "SIDH", "8483774855", "Y")
+		with self.assertRaises(ValueError):
+			fellow_1.save_state()
+
+	def test_save_state_phone_exists(self):
+		fellow = Fellow("LIOLZ", "SKIDH", "8483774855", "Y")
+		initial_person_count = len(State.persons)
+		fellow.save_state()
+		fellow_1 = Fellow("LIOZ", "SIDH", "8483874855", "Y")
```
Comment (mnjaggah) [inline fellow_tests.py]: typo: phone numbers should be the same

**Comment by mnjaggah** [comment]
Really good testcases

**Review by jimmykimani — COMMENTED** [review]
Your code looks fine  and well descriptive

File: fellow_tests.py (hunk 1 above)
Comment (GDP254) [reply fellow_tests.py]: Thanks, I completely missed that one.

File: add_person_tests.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from unittest import TestCase, main
+from fellow_tests import TestFellow
+from staff_tests import TestStaff
```
Comment (GDP254) [inline add_person_tests.py]: Good flow of order but spacing is missing. i.e

>  Imports should be grouped in the following order:
> 
>     standard library imports
>     related third party imports
>     local application/library specific imports
> 
> You should put a blank line between each group of imports. 
Source: https://www.python.org/dev/peps/pep-0008/#imports

File: add_person_tests.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+class TestAddPerson(TestCase):
+	def test_add_person_successfully(self):
+		TestFellow.test_save_state_new_phone()
+		TestStaff.test_save_state_new_phone()
```
Comment (GDP254) [inline add_person_tests.py]: This code runs all tests related to adding persons (repetitive) but it ignore others implied requirements from the specification. For example, testing interface messaging, testing scenarios for varied sources, testing scenarios for multiple entry.

File: fellow_tests.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+		new_person_count = len(State.persons)
+		self.assertEquals(initial_person_count, new_person_count)
+
+	def test_save_state_new_phone(self):
+		fellow = Fellow("Standard")
+		initial_person_count = len(State.persons)
+		fellow.save_state()
+		new_person_count = len(State.persons)
+		self.assertEquals(initial_person_count + 1, new_person_count)
+
```
Comment (GDP254) [inline fellow_tests.py]: Test cases are long, Grouping could help reader understand what is going on quicker.

File: models/room.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+Rooms are spaces within the Dojo.
+They can have unique names, capacity and two types
+ i.e. Office/ LivingSpace.
+"""
+
+class Room(object):
+
+	name = "Room"
+	capacity = 6
```
Comment (maquchizi) [inline models/room.py]: Do all rooms have a capacity of 6? Please confirm.

File: tests/fellow_tests.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+	def test_constructor_no_argument(self):
+		with self.assertRaises(ValueError):
+			fellow = Fellow()
+
+	def test_constructor_empty_string(self):
+		with self.assertRaises(ValueError):
+			fellow = Fellow("","","","")
+
+	def test_constructor_none(self):
```
Comment (maquchizi) [inline tests/fellow_tests.py]: This is, in effect, similar to empty_string and no_argument when instantiating a object. I don't this it is necessary. Do you agree?

File: tests/staff_tests.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+	def test_constructor_positive_float(self):
+		with self.assertRaises(ValueError):
+			staff = Staff(4.3, "Direde", "0794838434")
+
+	def test_constructor_negative_integer(self):
+		with self.assertRaises(ValueError):
+			staff = Staff(-43, "Direde", "0794838434")
+
+	def test_constructor_negative_float(self):
```
Comment (maquchizi) [inline tests/staff_tests.py]: Again, no need to test for both negative and positive integers/float. Just one will suffice.

**Review by maquchizi — APPROVED** [review]
Good work with the tests! One thing that I noted is that you have repetitive tests. Please try to keep the tests minimal, you can always refactor to cover edge cases as you go. 

Please merge this branch to master to close this merge request. Work on another branch for the next task.

File: models/room.py (hunk 5 above)
Comment (GDP254) [reply models/room.py]: No they do not. I put 6 given that it is the max capacity among the current room types and I needed a default value. Subclasses Office and LivingSpace have 4 and 6 respectively.

File: tests/fellow_tests.py (hunk 6 above)
Comment (GDP254) [reply tests/fellow_tests.py]: I agree. It isn't necessary. I will remove it.

File: tests/staff_tests.py (hunk 7 above)
Comment (GDP254) [reply tests/staff_tests.py]: I agree. It is unnecessary.
